linked-container/src/test/java/com/example/linkedcontainer/RedmineContainer.java

```diff
@@ -0,0 +1,44 @@
+package com.example.linkedcontainer;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.traits.LinkableContainer;
+import org.testcontainers.containers.wait.Wait;
+
+/**
+ * A linkable Redmine container.
+ */
+public class RedmineContainer<SELF extends RedmineContainer<SELF>> extends GenericContainer<SELF> implements LinkableContainer {
+
+    private static final int REDMINE_PORT = 3000;
+
+    public RedmineContainer(String dockerImageName) {
+        super.setDockerImageName(dockerImageName);
+    }
+
+    @Override
+    protected Integer getLivenessCheckPort() {
+        return getMappedPort(REDMINE_PORT);
+    }
+
+    @Override
+    protected void configure() {
+        addExposedPort(REDMINE_PORT);
+        waitingFor(Wait.forHttp("/"));
+    }
+
+    @Override
+    public SELF withEnv(String key, String value) {
```

Comment:
That has simplified it somewhat. Thanks!
I was trying to copy the way it had been done in `BrowserWebDriverContainer` but it doesn't need to be as complicated as that for this example.